ega-data-api-dataedge/src/main/java/eu/elixir/ega/ebi/dataedge/service/internal/EBINuFileService.java

```diff
@@ -0,0 +1,194 @@
+ *
+ */
+package eu.elixir.ega.ebi.dataedge.service.internal;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+import eu.elixir.ega.ebi.commons.exception.NotFoundException;
+import eu.elixir.ega.ebi.commons.shared.dto.File;
+import eu.elixir.ega.ebi.commons.shared.service.FileInfoService;
+import eu.elixir.ega.ebi.dataedge.exception.EgaFileNotFoundException;
+import eu.elixir.ega.ebi.dataedge.exception.FileNotAvailableException;
+import eu.elixir.ega.ebi.dataedge.exception.RangesNotSatisfiableException;
+import eu.elixir.ega.ebi.dataedge.exception.UnretrievableFileException;
+import eu.elixir.ega.ebi.dataedge.service.KeyService;
+import eu.elixir.ega.ebi.dataedge.service.NuFileService;
+import eu.elixir.ega.ebi.dataedge.utils.DecryptionUtils;
+import htsjdk.samtools.seekablestream.cipher.ebi.Glue;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.io.input.BoundedInputStream;
+import uk.ac.ebi.ega.fire.exceptions.ClientProtocolException;
+import uk.ac.ebi.ega.fire.exceptions.FireServiceException;
+import uk.ac.ebi.ega.fire.service.IFireService;
+
+import javax.crypto.Cipher;
+import javax.crypto.CipherInputStream;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.IvParameterSpec;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.concurrent.TimeUnit;
+
+public class EBINuFileService implements NuFileService {
+
+    private final KeyService keyService;
+    private final FileInfoService fileInfoService;
+    private final IFireService fireClientService;
+    private final LoadingCache<String, byte[]> aesHeaderCache;
+
+    public EBINuFileService(KeyService keyService, FileInfoService fileInfoService, IFireService fireClientService) {
+        this.keyService = keyService;
+        this.fileInfoService = fileInfoService;
+        this.fireClientService = fireClientService;
+        this.aesHeaderCache = Caffeine.newBuilder()
+                .expireAfterAccess(1, TimeUnit.DAYS)
+                .build(this::getAESHeader);
+    }
+
+    @Override
+    public long getPlainFileSize(String fileId) throws EgaFileNotFoundException, UnretrievableFileException, FileNotAvailableException {
+
+        return getPlainFileSize(getFile(fileId), keyService.getEncryptionAlgorithm(fileId));
+    }
+
+    private File getFile(String fileId) throws EgaFileNotFoundException, FileNotAvailableException {
+        File file;
+        try {
+            file = fileInfoService.getFileInfo(fileId);
+        } catch (NotFoundException exception) {
+            throw new EgaFileNotFoundException(fileId);
+        }
+
+        if (!file.getFileStatus().equals("available"))
```

Comment:
Can be written as
`!"available".equals(file.getFileStatus())`
this way null exceptions can be avoided